apps/readest-app/src-tauri/src/lib.rs

```diff
@@ -225,3 +248,83 @@ pub fn run() {
             },
         );
 }
+
+fn validate_book_file(path: &str) -> Result<(), String> {
+    // Basic file validation
+    if !Path::new(path).exists() {
+        return Err("File does not exist".to_string());
+    }
+
+    // Check if file is readable
+    match fs::metadata(path) {
+        Ok(metadata) => {
+            if metadata.len() == 0 {
+                return Err("File is empty".to_string());
+            }
+        }
+        Err(e) => return Err(format!("Failed to read file metadata: {}", e)),
+    }
+
+    Ok(())
+}
+
+#[command]
+async fn find_book_files(path: String) -> Result<Vec<String>, String> {
+    let book_extensions: Vec<&str> = vec!["epub", "pdf", "mobi", "azw3", "txt"];
```

Comment:
same list has been defined in `readest-app/src/services/constants.ts`, better to consolidate for a single source of truth of supported extensions

Reply:
the issue is that i'm not sure how to set this as a single source between rust and js though

Reply:
Can we pass the list from JavaScript to rust as the payload of the invoke?